modules/bmv/bucket.go

```diff
@@ -0,0 +1,75 @@
+package bmv
+
+import (
+	"sync"
+	"time"
+)
+
+// bucket is responsible for all messages between its time and the next bucket
+type bucket struct {
+	time time.Time
+	mtx  sync.RWMutex
+	data map[[32]byte]time.Time
+}
+
+func newBucket() *bucket {
+	b := new(bucket)
+	b.data = make(map[[32]byte]time.Time)
+	return b
+}
+
+// Set assigns the given time to the given key
+func (b *bucket) Set(key [32]byte, time time.Time) {
+	b.mtx.Lock()
+	defer b.mtx.Unlock()
+	b.data[key] = time
+}
+
+// Get returns the time attached to the key and true, or zero time and false
+// if the key was not present
+func (b *bucket) Get(key [32]byte) (time.Time, bool) {
+	b.mtx.RLock()
+	defer b.mtx.RUnlock()
+	t, ok := b.data[key]
```

Comment:
don't need "t, ok" as intermediary variables

Reply:
Unfortunately we do, since go's compiler isn't smart enough to figure this one out: https://play.golang.org/p/MC64njwnIT5